core/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java

```diff
@@ -55,19 +54,16 @@ public OperationRouting(Settings settings, AwarenessAllocationDecider awarenessA
     }
 
     public ShardIterator indexShards(ClusterState clusterState, String index, String type, String id, @Nullable String routing) {
```

Comment:
it's used in 2.x and asked Areek to keep it to limit the scope of the change as I hope we can back port it.